eppo/src/main/java/cloud/eppo/android/EppoClient.java

```diff
@@ -128,17 +128,80 @@ public String getAssignment(String subjectKey, String flagKey, SubjectAttributes
             return null;
         }
 
-        Variation assignedVariation = getAssignedVariation(subjectKey, flagKey, flag.getSubjectShards(), allocation.getVariations());
+        Variation assignedVariation = getAssignedVariation(subjectKey, flagKey, flag.getSubjectShards(),
+                allocation.getVariations());
         if (assignedVariation == null) {
             return null;
         }
 
         if (assignmentLogger != null) {
-            Assignment assignment = new Assignment(flagKey, assignedVariation.getValue(), subjectKey, Utils.getISODate(new Date()), subjectAttributes);
+            Assignment assignment = new Assignment(flagKey,
+                    assignedVariation.getTypedValue().stringValue(), subjectKey,
+                    Utils.getISODate(new Date()), subjectAttributes);
             assignmentLogger.logAssignment(assignment);
         }
 
-        return assignedVariation.getValue();
+        return assignedVariation.getTypedValue();
+    }
+
+    public String getAssignment(String subjectKey, String flagKey, SubjectAttributes subjectAttributes) {
+        return this.getStringAssignment(subjectKey, flagKey, subjectAttributes);
+    }
+
+    public String getAssignment(String subjectKey, String flagKey) {
+        return this.getStringAssignment(subjectKey, flagKey);
+    }
+
+    public String getStringAssignment(String subjectKey, String flagKey, SubjectAttributes subjectAttributes) {
+        EppoValue value = this.getTypedAssignment(subjectKey, flagKey, subjectAttributes);
+        if (value == null) {
+            return null;
+        }
```

Comment:
Mostly a style question in general - some of the other SDKs have a `Null` enum that the `EppoValue` can be - personally seems easier just to check against `null` like you do here. Any advantage that you see?

Reply:
Agreed thank you - I avoided it in the js changes https://github.com/Eppo-exp/js-client-sdk-common/pull/9